Bump sdk to v1.4.1

BaptisteGi opened the pull request.

FragmentedPacket commented on pyproject.toml:
```diff
@@ -40,6 +40,7 @@ pytest-asyncio = "^0.21.1"
 types-python-slugify = "*"
 invoke = "2.2.0"
 yamllint = "*"
+setuptools = "^75.7.0"
```
Is this added to work with Python 3.12?

BaptisteGi replied:
I think all versions of python were impacted here. From my understanding it's pytest that needs a module called `pkg_resources` to work: https://github.com/opsmill/nornir-infrahub/actions/runs/12631013053/job/35191943359\

(and `pkg_resources` is part of setuptools)

wvandeun commented on nornir_infrahub/plugins/inventory/infrahub.py:
```diff
@@ -7,7 +7,7 @@
 from infrahub_sdk import Config, InfrahubClientSync
 from infrahub_sdk.node import InfrahubNodeSync
-from infrahub_sdk.schema import NodeSchema
+from infrahub_sdk.schema import NodeSchemaAPI
```
Doing this change, means we have to update the min version of the infrahub-sdk dependency.
Also we'll need to make a release and therefor bump the package version.

wvandeun commented on nornir_infrahub/plugins/tasks/artifact.py:
```diff
@@ -4,7 +4,9 @@
     node = task.host.data["InfrahubNode"]
-    artifact_node = node._client.get(kind="CoreArtifact", name__value=artifact, object__ids=[node.id])
+    artifact_node = node._client.get(
+        kind="CoreArtifact", definition__artifact_name__value=artifact, object__ids=[node.id]
```
What is the reason behind this change?

FragmentedPacket replied:
We're passing in the name of the artifact definition, at least in our example. I may have made this change hastily, and I should have updated the `examples/nornir_tasks.py` to pass in the artifact name rather than the artifact definition name.

wvandeun replied:
We can change the behavior for sure if that is desirable, but this would be a breaking change that we need to communicate properly

wvandeun commented:
@FragmentedPacket  not sure about the try/except logic. We would need to this everywhere we do that import (tests) as well. Maybe we should just set the min version of the SDK. 

I think, for now this should be ok. I don't think we want to maintain to much of a backwards compatibility anyway. Things are still changing quite a bit anyway. So this shouldn't be to big of a problem.

wvandeun committed:
add setuptools as dev dependency

https://github.com/nornir-automation/nornir/issues/917
nornir should add this as a dependency which will happen in the new
3.5.0 release

wvandeun committed:
bump infrahub-sdk to v1.4.1 and set min version to v1.3.0

v1.3.0 introduces new pydantic models for different schema related
models that we need to use in this project

wvandeun committed:
use new NodeSchemaAPI model introduced in infrahub-sdk 1.3.0

wvandeun committed:
use setup-python@v5 in ci

wvandeun committed:
set PIPX_DEFAULT_PYTHON in CI

wvandeun committed:
install dev dependencies in CI

wvandeun committed:
update tasks example

wvandeun committed:
bump package version to 1.0.1
